### PR TITLE
[SW-645] Improve UX regarding 30 sec. delay until vault is active in the wallet

### PR DESCRIPTION
**Pull request by pstachel**

### CODE REVIEW

File: src/types/index.ts [hunk 1]
```diff
@@ -178,6 +173,7 @@ export interface IMultisigAccount extends IMultisigConsensus, IMultisigAccountRe
   signers: string[];
   consensusLabel?: string;
   hasPendingTransaction: boolean;
+
```
Comment (CedrikNikita) [inline src/types/index.ts:176]:
```suggestion
```

File: src/composables/multisigAccountCreate.ts [hunk 2]
```diff
@@ -163,17 +182,43 @@ export function useMultisigAccountCreate({ store }: IDefaultComposableOptions) {
...
     }
 
-    // Wait for the account to be discovered by the wallet to allow to open it's details.
-    // TODO in the future we could try to add the new account to the list of the accounts.
+    addPendingMultisigAccount(multisigAccount.value);
+    setActiveMultisigAccountId(multisigAccount.value.gaAccountId);
+
+    const delay = () => new Promise(() => setTimeout(() => {}, 10000));
+
+    await delay();
```
Comment (peronczyk) [inline src/composables/multisigAccountCreate.ts:220]: This looks rather bad. At least add comment that would describe why we need to wait 10s. Also the delay function should be moved to utils in my opinion.

File: src/composables/multisigAccountCreate.ts [hunk 3]
```diff
@@ -163,17 +182,43 @@ export function useMultisigAccountCreate({ store }: IDefaultComposableOptions) {
...
         contractId: gaContract.contractId,
-        multisigAccountId: accountId,
+        balance: new BigNumber(gaContract.balance),
+        gaAccountId: accountId,
+        signers,
+        confirmationsRequired,
+        consensusLabel: prepareConsensusLabel(0, confirmationsRequired, signers.length),
+        nonce: 1, // Default value for freshly created account
+        createdAt: dayjs().toISOString(),
+        updatedAt: dayjs().toISOString(),
```
Comment (peronczyk) [inline src/composables/multisigAccountCreate.ts:195]: The `dayjs()` function is fired twice - I suggest to create const for this before the assignment.

File: src/composables/multisigAccountCreate.ts [hunk 4]
```diff
@@ -1,15 +1,21 @@
 import { computed, ref } from '@vue/composition-api';
 import BigNumber from 'bignumber.js';
 import { AmountFormatter, Crypto, TxBuilder } from '@aeternity/aepp-sdk';
+import dayjs from 'dayjs';
 import type {
   IDefaultComposableOptions,
-  IKeyPair,
-  IMultisigAccountBase,
+  IKeyPair, IMultisigAccount,
```
Comment (peronczyk) [inline src/composables/multisigAccountCreate.ts:7]: Please split this line.

File: src/composables/multisigAccounts.ts [hunk 5]
```diff
@@ -148,10 +182,10 @@ export function useMultisigAccounts({ store }: IDefaultComposableOptions) {
...
             const hasPendingTransaction = !!txHash && !consensus.expired;
 
-            const consensusLabel = (
-              `${hasPendingTransaction ? consensus?.confirmedBy?.length : 0}`
-              + `/${consensus.confirmationsRequired} ${i18n.t('common.of')} `
-              + `${signers.decodedResult?.length}`
+            const consensusLabel = prepareConsensusLabel(
+              hasPendingTransaction ? consensus?.confirmedBy?.length : 0,
+              consensus.confirmationsRequired,
+                signers.decodedResult?.length,
```
Comment (peronczyk) [inline src/composables/multisigAccounts.ts:188]: From the beginning this should not be placed here as it saves the outcome of the translation function to the object property. So when user changes the language this label won't change in some places.
So the `prepareConsensusLabel` function should be used directly in the templates.

**Review by peronczyk — CHANGES_REQUESTED** [review]

File: src/composables/multisigAccountCreate.ts (hunk 2 above)
Comment (pstachel) [reply src/composables/multisigAccountCreate.ts:220]: Ooops, forgot to delete it

File: src/popup/components/AccountCardMultisig.vue [hunk 6]
```diff
@@ -32,9 +34,12 @@ import AccountCardConsensus from './AccountCardConsensus.vue';
 import AccountCardBase from './AccountCardBase.vue';
 
 import type { IMultisigAccount } from '../../types';
+import AccountCardSyncing from './AccountCardSyncing.vue';
```
Comment (peronczyk) [inline src/popup/components/AccountCardMultisig.vue:37]: Please put this at the end.

File: src/popup/components/MultisigVaultCreateProgress.vue [hunk 7]
```diff
@@ -132,6 +145,14 @@ export default defineComponent({
       return index === currentPhaseIndex.value + 1;
     }
 
+    function currentPhaseText(index: number, localPhase: PhaseLabel) {
```
Comment (peronczyk) [inline src/popup/components/MultisigVaultCreateProgress.vue:148]: In my opinion this function is not necessary as this logic would still makes sense on the template side:

```
<div v-if="isPhaseCurrent(index)">
  {{ localPhase.pendingText || localPhase.text }}&hellip;
</div>
<div v-else>
  {{ localPhase.text }}
</div>
```
What do you think?

File: src/popup/utils/utils.ts [hunk 8]
```diff
@@ -443,3 +443,11 @@ export function getAeFee(value: number | string) {
...
   window.open(url, '_blank');
 }
+
+export function prepareConsensusLabel(
+  confirmedBy: number,
+  confirmationsRequired: number,
+  totalConfirmations: number,
+) {
+  return `${confirmedBy}/${confirmationsRequired} ${i18n.t('common.of')} ${totalConfirmations}`;
+}
```
Comment (peronczyk) [inline src/popup/utils/utils.ts:453]: In my opinion this should be a component.

File: src/popup/components/AccountCardMultisig.vue [hunk 9]
```diff
@@ -15,12 +15,14 @@
       <AccountCardConsensus
         :multisig-account="account"
       />
+      <AccountCardSyncing v-if="isPendingAccount" />
```
[the remark this answers is not in the document]
Comment (pstachel) [reply src/popup/components/AccountCardMultisig.vue:18]: They won't overlap but I think your suggestion will make it obvious - good catch

File: src/popup/components/MultisigVaultCreateProgress.vue (hunk 7 above)
Comment (pstachel) [reply src/popup/components/MultisigVaultCreateProgress.vue:148]: of course, I think that's just a cosmetic change - I prefer less elements in the template

File: src/composables/multisigAccounts.ts [hunk 10]
```diff
@@ -49,8 +50,14 @@ export function useMultisigAccounts({ store }: IDefaultComposableOptions) {
 
   const accounts = computed<IAccount[]>(() => store.getters.accounts);
   const activeNetwork = computed<INetwork>(() => store.getters.activeNetwork);
-  const activeMultisigAccount = computed<IMultisigAccount | undefined>(() => multisigAccounts.value
-    .find((account) => account.gaAccountId === activeMultisigAccountId.value));
+  const allMultisigAccounts = computed<IMultisigAccount[]>(
+    () => [...multisigAccounts.value, ...pendingMultisigAccounts.value],
+  );
+
+  const activeMultisigAccount = computed<IMultisigAccount | undefined>(
```
Comment (konradodo) [inline src/composables/multisigAccounts.ts:57]: activeMultisigAccount can be undefined? if not you can change te type and add on end of "find" "!"

File: src/composables/multisigAccounts.ts [hunk 11]
```diff
@@ -60,14 +67,41 @@ export function useMultisigAccounts({ store }: IDefaultComposableOptions) {
...
+    pendingMultisigAccounts.value = pendingMultisigAccounts.value.map((account) => {
+      if (account.gaAccountId === gaAccountId) {
+        return {
+          ...account,
+          txHash,
+          hasPendingTransaction: true,
+        };
+      }
+      return account;
+    });
```
Comment (konradodo) [inline src/composables/multisigAccounts.ts:92]:
```suggestion

      return account.gaAccountId === gaAccountId
        ? {
          ...account,
          txHash,
          hasPendingTransaction: true,
        }
       : account
```

File: src/popup/components/AccountCardMultisig.vue [hunk 12]
```diff
@@ -44,8 +49,17 @@ export default defineComponent({
...
   },
-  setup() {
+  setup(props, { root }) {
+    const { pendingMultisigAccounts } = useMultisigAccounts({ store: root.$store });
+
+    const isPendingAccount = computed(
+      () => pendingMultisigAccounts.value.find(
+        ({ gaAccountId }) => gaAccountId === props.account.gaAccountId,
+      ),
+    );
```
Comment (konradodo) [inline src/popup/components/AccountCardMultisig.vue:59]: isPendingAccount -> this return account, not boolean

File: src/popup/components/AccountCardSyncing.vue [hunk 13]
```diff
@@ -0,0 +1,42 @@
+<template>
+  <div class="account-card-syncing">
+    <PendingIcon class="pending-icon" />
+    <span class="message">{{ $t('modals.creatingMultisigAccount.syncing') }}</span>
```
Comment (konradodo) [inline src/popup/components/AccountCardSyncing.vue:4]: split line

**Review by konradodo — APPROVED** [review]

File: src/composables/multisigAccounts.ts (hunk 10 above)
Comment (pstachel) [reply src/composables/multisigAccounts.ts:57]: yes

File: src/composables/multisigAccounts.ts (hunk 5 above)
Comment (pstachel) [reply src/composables/multisigAccounts.ts:188]: I'll create a task for it as it already existed in our current code

File: src/popup/utils/utils.ts (hunk 8 above)
Comment (pstachel) [reply src/popup/utils/utils.ts:453]: I'll create a task for it with the refactor (which I tried to avoid in this PR) of all places where it exists

**Review by peronczyk — APPROVED** [review]

**Pull request merged**